lib/install/validate/validator.go

```diff
@@ -559,107 +558,107 @@ func (v *Validator) registries(ctx context.Context, input *data.Data, conf *conf
+	insecureRegistriesSet := dynamic.ParseRegistries(input.InsecureRegistries)
+
+	// Test insecure registries' reachability
+	for _, r := range insecureRegistriesSet {
+		r, ok := r.(registry.URLEntry)
+		if !ok {
+			err = fmt.Errorf("invalid insecure registry entry: %s", r)
+			v.NoteIssue(err)
+			return nil, nil, err
+		}
+
+		// Remove intersection between insecure registries and whitelist registries from whitelist set so
+		// we can ensure we test the exclusion set with certs
+		found := false
+		for idx, s := range secureRegistries {
+			if s.IsURL() && r.Match(s.String()) {
+				// remove the insecure registry from list of registries to get validated against certs
+				secureRegistries = append(secureRegistries[:idx], secureRegistries[idx+1:]...)
+				found = true
 				break
 			}
 		}
 
-		// remove the insecure registry from list of registries to get validated against certs
-		if exist {
-			secureRegistries = append(secureRegistries[:idx], secureRegistries[idx+1:]...)
+		if found {
+			continue
 		}
-	}
 
-	// Test insecure registries' reachability
-	for _, r := range input.InsecureRegistries {
 		// Make sure address is not a wildcard domain or CIDR.  If it is, do not validate.
-		if strings.HasPrefix(r.Host, "*") {
-			log.Debugf("Skipping registry validation for %s", r.Host)
+		if strings.HasPrefix(r.URL().Host, "*") {
+			log.Debugf("Skipping registry validation for %s", r)
 			continue
 		}
-		if _, _, err = net.ParseCIDR(r.Host + r.Path); err == nil {
-			log.Debugf("Skipping registry validation for %s%s", r.Host, r.Path)
-			continue
+
+		schemes := []string{""}
+		if r.URL().Scheme == "" {
+			schemes = []string{"https", "http"}
 		}
 
-		_, err = registryutils.Reachable(r.Host, "https", "", "", nil, registryValidationTime, true)
-		if err != nil {
-			_, err = registryutils.Reachable(r.Host, "http", "", "", nil, registryValidationTime, true)
+		rs := r.String()
+		for _, s := range schemes {
+			if _, err = registry.Reachable(rs, s, "", "", nil, registryValidationTime, true); err == nil {
+				break
+			}
 		}
 
 		if err != nil {
-			log.Warnf("Unable to confirm insecure registry %s is a valid registry at this time.", r.Host)
+			log.Warnf("Unable to confirm insecure registry %s is a valid registry at this time.", r)
 		} else {
-			log.Debugf("Insecure registry %s confirmed.", r.Host)
+			log.Debugf("Insecure registry %s confirmed.", r)
 		}
 	}
 
 	// Test secure registries' reachability
 	for _, w := range secureRegistries {
 		// Make sure address is not a wildcard domain or CIDR.  If it is, do not validate.
-		if strings.HasPrefix(w.Hostname(), "*") {
-			log.Debugf("Skipping registry validation for %s", w.Host)
+		if w.IsCIDR() {
+			log.Debugf("Skipping registry validation for %s", w)
 			continue
 		}
-		if _, _, err = net.ParseCIDR(w.Host + w.Path); err == nil {
-			log.Debugf("Skipping registry validation for %s%s", w.Host, w.Path)
+
+		w, ok := w.(registry.URLEntry)
+		if !ok {
```

Comment:
Like above, I'm not sure you'll ever get !ok as ParseRegistries throws away non URLEntry.